app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,56 @@
+class TripsController < ApplicationController
+  def index
+    @trips = Trip.all
+  end
+
+  def show
+    id = params[:id].to_i
+    @trip = Trip.find_by(id: id)
+
+
+    if @trip.nil?
+      render :notfound, status: :not_found
+    end
+  end
+
+  def edit
+    @trip = Trip.find(params[:id].to_i)
+  end
+
+  def update
+      @trip = Trip.find_by(id: params[:id].to_i)
+    if  @trip.update(trip_params)
+      redirect_to trip_path(@trip.id)
+    else
+      render :edit
+    end
+  end
+
+  def new
+    @trip = Trip.new
+  end
+
+  def create
+    available_drivers = Driver.where(status: true)
+    offset = rand(available_drivers.count)
+    driver = available_drivers.offset(offset).first
+    new_trip = {driver_id: driver.id, passenger_id: params[:passenger_id].to_i, date: DateTime.now, cost: rand(500..3000),rating: 1}
```

Comment:
So new trips always start with a rating of 1?  That would make for a lot of unhappy drivers.

Reply:
When the trip ends, the user must rate the trip.